game-engine/src/test/scala/Winning_State_Tests.scala

```diff
@@ -0,0 +1,108 @@
+    val testCarGamePlayer1 = testGamePlayer1.asInstanceOf[CarGamePlayer]
+    TestHelper.putPlayerSomewhereOnTheTrack(carGameMap, testCarGamePlayer1.getGamePlayerId(), 1, 3)
+    testCarGamePlayer1.accelerate()
+
+    val testGamePlayer2 = TestHelper.getTestGamePlayer2()
+    val testCarGamePlayer2 = testGamePlayer2.asInstanceOf[CarGamePlayer]
+    TestHelper.putPlayerSomewhereOnTheTrack(carGameMap, testCarGamePlayer2.getGamePlayerId(), 2, 3)
+
+    TestHelper.processRound(gameMap, nothingCommand, nothingCommand)
+
+    val winner = carGameMap.getWinningPlayer.asInstanceOf[CarGamePlayer]
+    assert(testCarGamePlayer1.getGamePlayerId().equals(winner.getGamePlayerId()), "Player 1 was traveling at the fastest speed")
+  }
+
+  test("Given the match ends due to max rounds reached player 1 needs to win because of highest score") {
+    initialise()
+    val gameMap = TestHelper.initialiseGameWithNoMapObjects()
+    val carGameMap = gameMap.asInstanceOf[CarGameMap]
+    carGameMap.setCurrentRound(Config.MAX_ROUNDS)
+
+    val testGamePlayer1 = TestHelper.getTestGamePlayer1()
+    val testCarGamePlayer1 = testGamePlayer1.asInstanceOf[CarGamePlayer]
+    TestHelper.putPlayerSomewhereOnTheTrack(carGameMap, testCarGamePlayer1.getGamePlayerId(), 1, 3)
+    testCarGamePlayer1.score = 10
+
+    val testGamePlayer2 = TestHelper.getTestGamePlayer2()
+    val testCarGamePlayer2 = testGamePlayer2.asInstanceOf[CarGamePlayer]
+    TestHelper.putPlayerSomewhereOnTheTrack(carGameMap, testCarGamePlayer2.getGamePlayerId(), 2, 3)
+
+    TestHelper.processRound(gameMap, nothingCommand, nothingCommand)
+
+    val winner = carGameMap.getWinningPlayer.asInstanceOf[CarGamePlayer]
+    assert(testCarGamePlayer1.getGamePlayerId().equals(winner.getGamePlayerId()), "Player 1 has the highest score")
+  }
+
+  test("Given the match ends due to max rounds reached player 1 needs to win because of furthest on track, fastest speed and highest score") {
+    initialise()
+    val gameMap = TestHelper.initialiseGameWithNoMapObjects()
+    val carGameMap = gameMap.asInstanceOf[CarGameMap]
+    carGameMap.setCurrentRound(Config.MAX_ROUNDS)
+
+    val testGamePlayer1 = TestHelper.getTestGamePlayer1()
+    val testCarGamePlayer1 = testGamePlayer1.asInstanceOf[CarGamePlayer]
+    TestHelper.putPlayerSomewhereOnTheTrack(carGameMap, testCarGamePlayer1.getGamePlayerId(), 1, 10)
+    testCarGamePlayer1.accelerate()
+    testCarGamePlayer1.score = 10
+
+    val testGamePlayer2 = TestHelper.getTestGamePlayer2()
+    val testCarGamePlayer2 = testGamePlayer2.asInstanceOf[CarGamePlayer]
+    TestHelper.putPlayerSomewhereOnTheTrack(carGameMap, testCarGamePlayer2.getGamePlayerId(), 2, 3)
+
+    TestHelper.processRound(gameMap,nothingCommand, nothingCommand)
+
+    val winner = carGameMap.getWinningPlayer.asInstanceOf[CarGamePlayer]
+    assert(testCarGamePlayer1.getGamePlayerId().equals(winner.getGamePlayerId()), "Player 1 has the highest score")
```

Comment:
```suggestion
    assert(testCarGamePlayer1.getGamePlayerId().equals(winner.getGamePlayerId()), "Winner player id should be Player 1")
```
Remember those messages helps the developer debug their code when the unit test fails (gives them a clear indication about what is expected in this test case)